File: Div3-Group1/AddingSpacesToAString.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string answer="";
        int j=0;
        
        for(int i=0;i<s.length();i++)
        {
            if(j<spaces.size()&&i==spaces[j])
            {
                answer+=" ";
                answer+=s[i];
                j++;
            }
            else
                answer+=s[i];
        }
        return answer;
    }
};

File: leet-code-solutions/adding-spaces-to-a-string.py
class Solution(object):
    def addSpaces(self, s, spaces):
        """
        :type s: str
        :type spaces: List[int]
        :rtype: str
        """
        left=0
        answer=[]
        spaces = set(spaces)
        for i in range(len(s)):
            if i in spaces:
                answer.append(" ")
                answer.append(s[i])            
                # index=spaces.index(i+1)
                # if index+1 < len(spaces):
                #     spaces[index+1]+=1
            else:           
                answer.append(s[i])            
            
        return "".join(answer)
        